File: backend/Spring-Boot-Ecommerce/src/main/java/com/domiverse/Ecommerce/entity/OrderItem.java
package com.domiverse.Ecommerce.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "unit_price")
    private BigDecimal unitPrice;

    @Column(name = "quantity")
    private int quantity;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "order_id", nullable = false)
    @JsonBackReference("order-orderitem")
    private Order order;
}

File: frontend/angular-ecommerce/src/app/common/order-item.ts
import { CartItem } from './cart-item';
import { Order } from './order';
export class OrderItem {

    imageUrl: string = '';
    unitPrice: number = 0; 
    quantity: number = 0;
    productId: number = 0;

    constructor(CartItem: CartItem) {
        this.imageUrl = CartItem.imageUrl;
        this.unitPrice = CartItem.unitPrice;
        this.quantity = CartItem.quantity;
        this.productId = CartItem.id;
    }
}
